en/features/verification.md

```diff
@@ -0,0 +1,195 @@
+
+layout: post
+lang: en
+columns: 1
+id: bitcoin-core-verification
+title: Verification - Bitcoin Core Features
+breadcrumbs:
+  - features
+---
+> Imagine a scientist reading about an experimental result and then
+> repeating the experiment for herself. Doing so allows her to trust
+> the result without having to trust the original scientists.
+
+Bitcoin Core checks each block of transactions it receives to ensure
+that everything in that block is fully valid---allowing it to trust the
+block without trusting the miner who created it.
+
+This prevents miners from tricking Bitcoin Core users into accepting
+blocks that violate the 21 million bitcoin limit or which break other
+important rules.
+
+Users of other wallets don't get this level of security, so they can be
+tricked into accepting fabricated transactions or alternative block chains.
+
+Why take that risk if you don't have to? Bitcoin Core provides the best
+possible security against dishonest miners along with additional
+security against other easier attacks (see below for details).
+
+## How Verification Protects Your Bitcoins
+
+Light wallets and wallet services (such as exchanges) put your bitcoins
+at increased risk of being stolen. That risk may be acceptable for small
+values of bitcoin on mobile wallets, but is it what you want for your
+real wallet?
+
+{:.center}
+*Click the title of any attack below to learn more about it.*
+
+<table>
+<tr>
+  <th>Attack</th>
+  <th markdown="span">Wallet Service</th>
+  <th markdown="span">Light Wallet</th>
+  <th>Bitcoin Core</th>
+</tr>
+
+<tr>
+  <td><a href="/{{page.lang}}/features/details/direct-theft/">Direct theft</a></td>
+  <td class="bgred"></td>
+  <td class="bggreen"></td>
+  <td class="bggreen"></td>
+</tr>
+
+<tr>
+  <td><a href="/{{page.lang}}/features/details/bait-and-switch/">Bait and switch</a></td>
+  <td class="bgred"></td>
+  <td class="bgyellow"></td>
+  <td class="bggreen"></td>
+</tr>
+
+<tr>
+<td><a href="/{{page.lang}}/features/details/fabricated-transactions/">Fabricated transactions</a></td>
+  <td class="bgred"></td>
+  <td class="bgred"></td>
+  <td class="bggreen"></td>
+</tr>
+
+<tr>
+<td><a href="/{{page.lang}}/features/details/transaction-withholding/">Transaction withholding</a></td>
+  <td class="bgred"></td>
+  <td class="bgred"></td>
+  <td class="bggreen"></td>
+</tr>
+
+<tr>
+<td><a href="/{{page.lang}}/features/details/chain-hijacking/">Chain hijacking</a></td>
+  <td class="bgred"></td>
+  <td class="bgred"></td>
+  <td class="bggreen"></td>
+</tr>
+
+<tr>
+<td><a href="/{{page.lang}}/features/details/chain-rewrites/">Chain rewrites</a></td>
+  <td class="bgred"></td>
+  <td class="bgred"></td>
+  <td class="bgred"></td>
+</tr>
+
+</table>
+
+Note that although all programs---including Bitcoin Core---are
+vulnerable to chain rewrites, Bitcoin provides a defense mechanism: the
+more confirmations your transactions have, the safer you are. *There is
+no known decentralized defense better than that.*
+
```

Comment:
I think this is a bit alarmist for people who are likely to be misinformed.  There is no evidence that miners are near a majority attack level of hashrate, there are updates like StratumV2 in the works, and it doesn't affect bitcoins at rest.  I don't believe there has ever been a chain rewrite for the recommended 6 confirmations?

Reply:
I'm moving this to the chain rewrites page per one of your previous comments, so it should have more context there.  That said,

> There is no evidence that miners are near a majority attack level of hashrate

I don't understand this comment, miners always have a majority attack level of hashrate: 100% of present hashrate is always greater than the 50% need to be a majority.

> there are updates like StratumV2 in the works

That doesn't change the attack, it just increases the number of people who need to coordinate to pull it off.

> it doesn't affect bitcoins at rest.

Huh?  If you can sustain a majority of hashrate, you can rewrite every block on the chain except the genesis block.  Take any UTXO in your wallet and follow its coin ancestry back to the most recent ancestor that was the output of a generation transaction.  If that block is removed from the chain, its generation transaction is not valid and so your UTXO is also invalid.

> I don't believe there has ever been a chain rewrite for the recommended 6 confirmations?

An example of a six block reorg from July 2015 is provided on the chain rewrite page.

Reply:
> I'm moving this to the chain rewrites page per one of your previous comments, so it should have more context there. 

Great, thanks!

> miners always have a majority attack level of hashrate

Yes, of course.  I should have been more clear - I meant that there is no evidence of a malicious miner or group of malicious miners building an alternative valid chain.  StratumV2, when fully implemented, vastly increases the number of people needed to coordinate to pull off the attack, and is a great defense.


> If you can sustain a majority of hashrate, you can rewrite every block on the chain except the genesis block

Yes, but it is much more likely that a 51% attack will start from a recent fork in the chain, thereby bitcoins "at rest" with many confirmations will not be affected.

> An example of a six block reorg from July 2015 is provided on the chain rewrite page.

That was an invalid block chain, and was not accepted by full nodes.  I don't believe there has ever been a block reorg of 6 blocks or longer.

Reply:
>  I don't believe there has ever been a block reorg of 6 blocks or longer.

2013 chain fork? https://github.com/bitcoin/bips/blob/master/bip-0050.mediawiki
2010 value overflow incident?  https://en.bitcoin.it/wiki/Value_overflow_incident

Reply:
Yes, of course - thank you for the links.